cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -48,6 +48,16 @@ type MasterConfiguration struct {
 	APIServerCertSANs []string `json:"apiServerCertSANs"`
 	// CertificatesDir specifies where to store or look for all required certificates
 	CertificatesDir string `json:"certificatesDir"`
+
+	MasterCertificates *MasterCertificates `json:"masterCertificates,omitempty"`
+	// Public Addr or DNS
+	PublicAddress string `json:"publicAddress"`
+	//Masters Count
+	Count int `json:"count"`
+
+	//required for cloud providers. To make right tag for resources
+	ClusterName      string
```

Comment:
Sorry. Forgot about this. But there is some misunderstanding.
There are two structures:

- v1alpha1.MasterConfiguration
- kubeadm.MasterConfiguration

And kubeadm.MasterConfiguration used to read configuration from file. May be it will be better to remove kubeadm.MasterConfiguration or embed v1alpha1.MasterConfiguration to it.
I can add json annotation, but it will be new pull request.
 Or Is it legal to change current pull request?

@timothysc Were are you chatting about HA features and how am I able to join this discussion?